### ISSUE TITLE
Deprecated api used in gradle plugin

### ISSUE THREAD

**Issue by c0nstexpr** [diffplug/spotless#1773]
related code
https://github.com/diffplug/spotless/blob/8b1713d88bf575730da921c910da72718b5da4e0/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/JvmLang.java#L44

log output
> The Project.getConvention() method has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.2.1/userguide/upgrading_version_8.html#deprecated_access_to_conventions
The org.gradle.api.plugins.Convention type has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.2.1/userguide/upgrading_version_8.html#deprecated_access_to_conventions
The org.gradle.api.plugins.JavaPluginConvention type has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.2.1/userguide/upgrading_version_8.html#java_convention_deprecation

**Comment by Goooler** [comment]
We noted it at https://github.com/diffplug/spotless/pull/1618#discussion_r1149943892.

**Comment by c0nstexpr** [comment]
> @c0nstexpr looks like you are using Gradle 8.2.1, which version of Spotless are you using?
> 
> @Goooler it seems to me like after we merged #1618, warnings like this shouldn't happen because it would take the other branch. Is there any chance Gradle is parsing the class files to look for API even if it isn't called?

I use the [latest version](https://github.com/c0nstexpr/fishology/blob/dev/gradle/libs.versions.toml) Spotless. And I added `org.gradle.warning.mode=all` in my [gradle properties](https://github.com/c0nstexpr/fishology/blob/dev/gradle.properties)

**Comment by Goooler** [comment]
I don't see any warning related to Spotless after enabling `org.gradle.warning.mode=all`, see
- https://github.com/Goooler/AndroidOAuth2WebView/commit/69b9f28efe466de7283d03929538168a003d4fba
- https://scans.gradle.com/s/j5f2thlqkpc5q

**Comment by c0nstexpr** [comment]
> 

build scan means [this one](https://scans.gradle.com/s/kd3sk4lxotylq)?

**Comment by Goooler** [comment]
Yeah, but seems no `The Project.getConvention() method has been deprecated.` in your scan.

**Comment by c0nstexpr** [comment]
> Yeah, but seems no `The Project.getConvention() method has been deprecated.` in your scan.

yes, I notice that. But I try again gradle sync, it just happed

> Starting Gradle Daemon...
Gradle Daemon started in 1 s 345 ms
> Task :buildSrc:generateExternalPluginSpecBuilders UP-TO-DATE
> Task :buildSrc:extractPrecompiledScriptPluginPlugins UP-TO-DATE
> Task :buildSrc:compilePluginsBlocks UP-TO-DATE
> Task :buildSrc:generatePrecompiledScriptPluginAccessors UP-TO-DATE
> Task :buildSrc:generateScriptPluginAdapters UP-TO-DATE
> Task :buildSrc:pluginDescriptors UP-TO-DATE
> Task :buildSrc:processResources UP-TO-DATE
> Task :buildSrc:compileKotlin
> Task :buildSrc:compileJava NO-SOURCE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
> Task :buildSrc:inspectClassesForKotlinIC

> Configure project :core
Fabric Loom: 1.3.8
configuring fishology-core fabric mod project
The resolvable usage is already allowed on configuration ':core:clientRuntimeClasspath'. This behavior has been deprecated. This behavior is scheduled to be removed in Gradle 9.0. Remove the call to setCanBeResolved(true), it has no effect. Consult the upgrading guide for further information: https://docs.gradle.org/8.2.1/userguide/upgrading_version_8.html#redundant_configuration_usage_activation
	at Fabric_common_gradle$6.execute(fabric-common.gradle.kts:51)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
The resolvable usage is already allowed on configuration ':core:testModClientRuntimeClasspath'. This behavior has been deprecated. This behavior is scheduled to be removed in Gradle 9.0. Remove the call to setCanBeResolved(true), it has no effect. Consult the upgrading guide for further information: https://docs.gradle.org/8.2.1/userguide/upgrading_version_8.html#redundant_configuration_usage_activation
	at Fabric_common_gradle$7.execute(fabric-common.gradle.kts:60)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
remap jar archive path: D:\Documents\fishology\core\build\libs\fishology-core-0.0.1.jar

> Configure project :interact
Fabric Loom: 1.3.8
configuring fishology-interact fabric mod project
The resolvable usage is already allowed on configuration ':interact:clientRuntimeClasspath'. This behavior has been deprecated. This behavior is scheduled to be removed in Gradle 9.0. Remove the call to setCanBeResolved(true), it has no effect. Consult the upgrading guide for further information: https://docs.gradle.org/8.2.1/userguide/upgrading_version_8.html#redundant_configuration_usage_activation
	at Fabric_common_gradle$6.execute(fabric-common.gradle.kts:51)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
The resolvable usage is already allowed on configuration ':interact:testModClientRuntimeClasspath'. This behavior has been deprecated. This behavior is scheduled to be removed in Gradle 9.0. Remove the call to setCanBeResolved(true), it has no effect. Consult the upgrading guide for further information: https://docs.gradle.org/8.2.1/userguide/upgrading_version_8.html#redundant_configuration_usage_activation
	at Fabric_common_gradle$7.execute(fabric-common.gradle.kts:60)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
remap jar archive path: D:\Documents\fishology\interact\build\libs\fishology-interact-0.0.1.jar

> Task :prepareKotlinBuildScriptModel UP-TO-DATE
> Task :interact:generateDLIConfig
> Task :core:generateDLIConfig
> Task :core:generateLog4jConfig
> Task :interact:generateLog4jConfig
> Task :core:generateRemapClasspath
> Task :interact:generateRemapClasspath
> Task :interact:configureLaunch
> Task :core:configureLaunch
> Task :interact:downloadAssets
> Task :interact:configureClientLaunch
> Task :interact:ideaSyncTask
> Task :core:downloadAssets
> Task :core:configureClientLaunch
> Task :core:ideaSyncTask
The Project.getConvention() method has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.2.1/userguide/upgrading_version_8.html#deprecated_access_to_conventions
The org.gradle.api.plugins.Convention type has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.2.1/userguide/upgrading_version_8.html#deprecated_access_to_conventions
The org.gradle.api.plugins.JavaPluginConvention type has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.2.1/userguide/upgrading_version_8.html#java_convention_deprecation
Resolution of the configuration :core:detachedConfiguration1 was attempted from a context different than the project context. Have a look at the documentation to understand why this is a problem and how it can be resolved. This behavior has been deprecated. This will fail with an error in Gradle 9.0. For more information, please refer to https://docs.gradle.org/8.2.1/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.
Resolution of the configuration :interact:detachedConfiguration1 was attempted from a context different than the project context. Have a look at the documentation to understand why this is a problem and how it can be resolved. This behavior has been deprecated. This will fail with an error in Gradle 9.0. For more information, please refer to https://docs.gradle.org/8.2.1/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.
Resolution of the configuration :core:spotless865457264 was attempted from a context different than the project context. Have a look at the documentation to understand why this is a problem and how it can be resolved. This behavior has been deprecated. This will fail with an error in Gradle 9.0. For more information, please refer to https://docs.gradle.org/8.2.1/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.
Resolution of the configuration :core:spotless-1972429381 was attempted from a context different than the project context. Have a look at the documentation to understand why this is a problem and how it can be resolved. This behavior has been deprecated. This will fail with an error in Gradle 9.0. For more information, please refer to https://docs.gradle.org/8.2.1/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.
Resolution of the configuration :interact:spotless865457264 was attempted from a context different than the project context. Have a look at the documentation to understand why this is a problem and how it can be resolved. This behavior has been deprecated. This will fail with an error in Gradle 9.0. For more information, please refer to https://docs.gradle.org/8.2.1/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.
Resolution of the configuration :interact:spotless-1972429381 was attempted from a context different than the project context. Have a look at the documentation to understand why this is a problem and how it can be resolved. This behavior has been deprecated. This will fail with an error in Gradle 9.0. For more information, please refer to https://docs.gradle.org/8.2.1/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.

BUILD SUCCESSFUL in 37s
20 actionable tasks: 12 executed, 8 up-to-date

**Comment by Goooler** [comment]
They are the warning sources:
```
The Project.getConvention() method has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.2.1/userguide/upgrading_version_8.html#deprecated_access_to_conventions
	at org.gradle.api.internal.project.DefaultProject.getConvention(DefaultProject.java:593)
	at org.jetbrains.plugins.gradle.tooling.util.JavaPluginUtil.getJavaPluginConvention(JavaPluginUtil.java:13)
	at org.jetbrains.plugins.gradle.tooling.util.JavaPluginUtil.getSourceSetContainer(JavaPluginUtil.java:18)
...
```
[full.txt](https://github.com/diffplug/spotless/files/12297583/full.txt)

**Comment by nedtwigg** [comment]
For anyone else who was slow like me, the stacktraces are coming from the IntelliJ sync plugin, not Spotless. Looks like we did cause this warning in the past, @Goooler fixed it, and it is still fixed. I'm happy for the discussion to continue, but I'm going to close the issue for now because I think we got to the bottom of it. Thanks for sharing the info @c0nstexpr, and thanks for fixing it and tracking this one down too @Goooler.
